lib/tasks/fix_coord_mismatch.rake

```diff
@@ -54,32 +52,78 @@ def geocode_fields(result, address, address1, address2, address3)
+  geocode_name = Geocoder.search(result.institution.downcase.split('#').first.downcase)
+  if geocode_name.present?
+    text_search(result, geocode_name)
+  else
+    search_bad_addr(result, geocoded3)
+  end
+end
+
+def parse_add_fields(res, field)
+  field.nil? ? nil : "#{field}, #{res.city}, #{res.state}, #{res.zip}, #{res.country}"
+end
+
+def parse_address(res, field)
+  if field.present?
+    "#{field}, #{res.city}, #{res.physical_country}"
+  else
+    "#{res.city}, #{res.physical_country}"
+  end
+end
+
+def search_bad_addr(result, geocoded3)
+  # parses bad addresses and flags and issues
+  stopwords = %w[ave avenue road rd dr drive blvd pkwy st street parkway]
   stopwords_regex = /\b(#{Regexp.union(*stopwords).source})\b/i
   add_split = result.address.split(stopwords_regex).map(&:strip)[0..1].join(' ')
   new_address = [add_split, result.state, result.zip].compact.join(' ')
   geocoded = Geocoder.coordinates(new_address)
-  # geocoded.present? ? update_mismatch(result, geocoded) : update_mismatch(result, geocoded3)
+  update_bad_addr(result, geocoded, geocoded3)
+end
+
+def update_bad_addr(result, geocoded, geocoded3)
   if geocoded.present?
     update_mismatch(result, geocoded)
   else
     update_mismatch(result, geocoded3)
+    result.update(bad_address: true)
   end
 end
 
-def parse_add(res, address)
-  if address.present?
-    "#{address}, #{res.city}, #{res.state}, #{res.zip}, #{res.country}"
-  else
-    "#{res.city}, #{res.state}, #{res.zip}, #{res.country}"
+def text_search(result, geocode_name)
+  geocode_name.each do |geo|
+    next if ck_srch_fields(result, geo).blank?
+
+    upd_data = [ck_srch_fields(result, geo).first.data['lat'].to_f, ck_srch_fields(result, geo).first.data['lon'].to_f]
+    update_mismatch(result, upd_data)
   end
 end
 
-def parse_address(res, field)
-  if field.present?
-    "#{field}, #{res.city}, #{res.physical_country}"
+def ck_srch_fields(result, geo)
+  state = geo.data['address']['ISO3166-2-lvl4'].present? ? geo.data['address']['ISO3166-2-lvl4'].split('-').last : nil
+  city = geo.data['address']['city']
+  zip = geo.data['address']['postcode']
+  num = geo.data['address']['amenity'].present? ? geo.data['address']['amenity'].split(' ').last.to_i : nil
+  if (num.is_a? Numeric) && (num > 1)
+    return_num_text_search(result, geo, city, state, num)
   else
-    "#{res.city}, #{res.physical_country}"
+    return_text_search(result, geo, city, state, zip)
+  end
+end
+
+def return_num_text_search(result, geo, city, state, num)
+  text_search = []
+  add_num = result.institution.downcase.split('#').last.to_i
+  text_search << geo if city == result.city.titleize && state == result.state && num == add_num
+end
+
+def return_text_search(result, geo, city, state, zip)
+  text_search = []
+  vil = geo.data['address']['village']
+  if city == result.city.titleize || state == result.state || zip == result.zip || vil == result.city.titleize
```

Comment:
If this conditional isn't met should you return an empty array here instead of nil?